client/my-sites/add-ons/hooks/use-add-on-checkout-link.ts

```diff
@@ -0,0 +1,24 @@
+import { useSelector } from 'react-redux';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+/**
+ * Returns a function that will return a formatted checkout link for the given add-on and quantity.
+ * E.g. https://wordpress.com/checkout/<siteSlug>/<addOnSlug>>:-q-<quantity>
+ *
+ * @returns {Function} A function returnig a formatted checkout link for the given add-on and quantity
+ */
+
+const useAddOnCheckoutLink = (): ( ( addOnSlug: string, quantity?: number ) => string ) => {
+	const selectedSite = useSelector( getSelectedSite );
+
+	return ( addOnSlug: string, quantity?: number ): string => {
+		const checkoutLinkFormat = `/checkout/${ selectedSite?.slug }/${ addOnSlug }`;
```

Comment:
If `selectedSite` does not exist, this will create a URL like `/checkout/undefined/product`. Do you think we should do something different in that case? Maybe just `/checkout/product` (which should work well by showing a site picker first).

Reply:
Thanks! I added a check for this now, and it will return the site picker if no `selectedSite` is provided.